src/platform_impl/linux/wayland/event_loop.rs

```diff
@@ -492,7 +519,20 @@ impl SeatData {
                         self.sink.clone(),
                         self.store.clone(),
                         self.modifiers_tracker.clone(),
-                    ))
+                    ));
+
+                    self.relative_pointer =
+                        self.relative_pointer_manager_proxy.as_ref().map_or_else(
+                            || None,
+                            |manager| {
+                                super::pointer::implement_relative_pointer(
+                                    self.sink.clone(),
+                                    self.pointer.as_ref().unwrap(),
+                                    manager,
+                                )
+                                .ok()
+                            },
```

Comment:
Can't this be simplified using `Option::and_then` rather than `map_or_else`?

Reply:
Yes, next commit fixes it.